### PR TITLE
feat(QC): reset validation status during submission

### PR DESCRIPTION
**Pull request by kulgan**

### CODE REVIEW

File: sheepdog/transactions/upload/transaction.py [hunk 1]
```diff
@@ -285,6 +285,13 @@ def add_entity(self, doc):
             self.logger.exception(e)
             self.record_error('Unable to parse entity')
 
+    def reset_validation_status(self):
```
Comment (anmaxvl) [inline sheepdog/transactions/upload/transaction.py]: defining this inside `TransactionBase` class didn't work? so you don't have to do it in both places?

Comment (kulgan) [reply sheepdog/transactions/upload/transaction.py]: It did not seem like something that belongs to the transaction base, the methods in there are at the entity level, this is somewhat above the entity level. Maybe something else that both of them can extend but that seems like a bigger change

Comment (anmaxvl) [reply sheepdog/transactions/upload/transaction.py]: hm. yeah, we could throw in some sort of a mixin, which might be quicker?

File: sheepdog/transactions/upload/__init__.py [hunk 2]
```diff
@@ -41,6 +41,7 @@ def single_transaction_worker(transaction, *doc_args):
             transaction.parse_doc(*doc_args)
             transaction.flush()
             transaction.post_validate()
+            transaction.reset_validation_status()
```
Comment (anmaxvl) [inline sheepdog/transactions/upload/__init__.py]: so back to my question... does this take care of resetting the flag when nodes are deleted? or new file nodes created/updated? sheepdog's class hierarchy isn't fresh in my head...

Comment (kulgan) [reply sheepdog/transactions/upload/__init__.py]: No it does not, just added that to the deletion transaction handler

File: tests/integration/submission/test_endpoints.py [hunk 3]
```diff
@@ -453,6 +453,12 @@ def test_disallow_cross_project_references(client, pg_driver, cgci_blgsp, tcga_b
 
 
 def test_delete_entity(client, pg_driver, cgci_blgsp, submitter, indexd_client):
+
+    # ensure project qc flag is not setup for validation
+    with pg_driver.session_scope():
+        project = pg_driver.nodes().get("daa208a7-f57a-562c-a04a-7a7c77542c98")
+        assert project._sysan.get("is_validated", True) is True
```
Comment (anmaxvl) [inline tests/integration/submission/test_endpoints.py]: shouldn't this change?

File: sheepdog/transactions/transaction_base.py [hunk 4]
```diff
@@ -507,3 +508,10 @@ def write_transaction_log(self):
             tx_log.timestamp = timestamp
             if tx_log.documents:
                 tx_log.documents[0].response_json = self.json
+
+    def reset_validation_status(self):
+        project = utils.lookup_project(
+            self.db_driver, self.program, self.project
+        )
+        project._sysan["validation_attempted"] = False
```
Comment (anmaxvl) [inline sheepdog/transactions/transaction_base.py]: move this to globals maybe? just in case we decide to change the name later?

**Pull request merged**